src/chunk.c

```diff
@@ -4569,10 +4570,22 @@ add_foreign_table_as_chunk(Oid relid, Hypertable *parent_ht)
 	 * Then add dimension constraints for the chunk
 	 */
 	ts_dimension_slice_insert_multi(chunk->cube->slices, chunk->cube->num_slices);
-	ts_chunk_constraints_add_dimension_constraints(chunk->constraints, chunk->fd.id, chunk->cube);
 	/* check constraints are not automatically created for foreign tables.
 	 * See: ts_chunk_constraints_add_dimension_constraints.
+	 * Collect all the check constraints from the hypertable and add them to the
+	 * foreign table. Otherwise, cannot add as child of the hypertable (pg inheritance
+	 * code will error. Note that the name of the check constraint on the hypertable
+	 * and the foreign table chunk should match.
+	 * Also, the dimension constraint is not converted to a check constraint on the
+	 * foreign table chunk
```

Comment:
There should be no dimension constraints processed here as they are not present on the parent table only on individual chunks.

Reply:
We add a dummy dimension constraint on the OSM chunk (line 4588). This is metadata in the chunk catalog table (cannot create a chunk without this info). But this is not translated into a check constraint. Is that what you are referring to?

Reply:
Updated the comment in the source code.